hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AmqpMessageDispatcherServiceIntegrationTest.java

```diff
@@ -603,4 +611,70 @@ private static boolean isDownloadAndUpdateRequest(final DmfMultiActionElement mu
                 || multiActionElement.getTopic().equals(EventTopic.DOWNLOAD_AND_INSTALL);
     }
 
+    @Test
+    @Description("Verify payload of batch assignment download and install message.")
+    public void assertBatchAssignmentsDownloadAndInstall() {
+        assertBatchAssignmentsMessagePayload(BATCH_DOWNLOAD_AND_INSTALL);
+    }
+
+    @Test
+    @Description("Verify payload of batch assignments download only message.")
+    public void assertBatchAssignmentsDownloadOnly() {
+        assertBatchAssignmentsMessagePayload(BATCH_DOWNLOAD);
+    }
+
+    protected void assertBatchAssignmentsMessagePayload(final EventTopic topic) {
+        enableBatchAssignments();
+
+        final List<String> targets = Arrays.asList("batchCtrlID1", "batchCtrlID2", "batchCtrlID3");
+        for (int i = 0; i < targets.size(); i++) {
+            registerAndAssertTargetWithExistingTenant(targets.get(i), i+1);
+        }
+
+        final DistributionSet ds = testdataFactory.createDistributionSet();
+        testdataFactory.addSoftwareModuleMetadata(ds);
+
+        final DistributionSetAssignmentResult assignmentResult = assignDistributionSet(ds.getId(), targets,
```

Comment:
please remove assignment to local var as it is not used

Reply:
done